gen3/bin/kube-setup-argo.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+#
+
+source "${GEN3_HOME}/gen3/lib/utils.sh"
+gen3_load "gen3/gen3setup"
+gen3_load "gen3/lib/kube-setup-init"
+
+
+ctx="$(g3kubectl config current-context)"
+ctxNamespace="$(g3kubectl config view -ojson | jq -r ".contexts | map(select(.name==\"$ctx\")) | .[0] | .context.namespace")"
+
+
+function setup_argo_buckets {
+  local accountNumber
+  local environment
+  local policyFile="$XDG_RUNTIME_DIR/policy_$$.json"
+  cat > "$policyFile" <<EOF
+{
+   "Version":"2012-10-17",
+   "Statement":[
+      {
+         "Effect":"Allow",
+         "Action":[
+            "s3:PutObject",
+            "s3:GetObject"
+         ],
+         "Resource":"arn:aws:s3:::$bucketName/*"
+      }
+   ]
+}
+EOF
+
+  if ! accountNumber="$(aws sts get-caller-identity --output text --query 'Account')"; then
+    gen3_log_err "could not determine account numer"
+    return 1
+  fi
+  if ! environment="$(g3kubectl get configmap manifest-global -o json | jq -r .data.environment)"; then
+    gen3_log_err "could not determine environment from manifest-global - bailing out of argo setup"
+    return 1
+  fi
+
+  # try to come up with a unique but composable bucket name
+  bucketName="gen3-argo-${accountNumber}-${environment//_/-}"
+
+  if ! secret="$(g3kubectl get secret argo-s3-creds -n argo 2> /dev/null)"; then
+    gen3_log_info "setting up bucket $bucketName"
+
+    if aws s3 ls --page-size 1 "s3://${bucketName}" > /dev/null 2>&1; then
+      gen3_log_info "${bucketName} s3 bucket already exists"
+      # continue on ...
+    elif ! aws s3 mb "s3://${bucketName}"; then
+      gen3_log_err "failed to create bucket ${bucketName}"
+    fi
+
+
+    gen3_log_info "Creating IAM user argo-user"
```

Comment:
Can we add $vpc_name or some other identifier to the user, so we can have multiple per account?